LoopKitUI/Views/Settings Editors/CorrectionRangeOverridesEditor.swift

```diff
@@ -0,0 +1,285 @@
+//
+//  CorrectionRangeOverridesEditor.swift
+//  Loop
+//
+//  Created by Michael Pangburn on 5/15/20.
+//  Copyright © 2020 LoopKit Authors. All rights reserved.
+//
+
+import SwiftUI
+import HealthKit
+import LoopKit
+
+public struct CorrectionRangeOverrides: Equatable {
+    enum Preset: Hashable {
+        case preMeal
+        case workout
+    }
+
+    var ranges: [Preset: ClosedRange<HKQuantity>]
+
+    public init(preMeal: DoubleRange?, workout: DoubleRange?, unit: HKUnit) {
+        ranges = [:]
+        ranges[.preMeal] = preMeal?.quantityRange(for: unit)
+        ranges[.workout] = workout?.quantityRange(for: unit)
+    }
+
+    public var preMeal: ClosedRange<HKQuantity>? { ranges[.preMeal] }
+    public var workout: ClosedRange<HKQuantity>? { ranges[.workout] }
+}
+
+public struct CorrectionRangeOverridesEditor: View {
+    var initialValue: CorrectionRangeOverrides
+    var unit: HKUnit
+    var minValue: HKQuantity?
+    var save: (_ overrides: CorrectionRangeOverrides) -> Void
+    var sensitivityOverridesEnabled: Bool
+    let mode: PresentationMode
+
+    @State private var userDidTap: Bool = false
```

Comment:
`userDidTap` keeps track of whether or not the user has tapped yet. If the user hasn't tapped, then instructional content is displayed, otherwise it isn't